polls/views.py

```diff
@@ -79,18 +81,48 @@ def post(self, request):
 
 class TokenView(LoginRequiredMixin, UserPassesTestMixin, View):
     def get(self, request):
-        admin_api_url = request.build_absolute_uri("/polls/api/v1/admin-api/")
+        admin_api = request.build_absolute_uri(reverse("admin-api"))
+        annotation_api = request.build_absolute_uri(reverse("annotation-api"))
+        lock_users_api = request.build_absolute_uri(reverse("lock-users-api"))
         token, _ = Token.objects.get_or_create(user=request.user)
-        context = {"token": token, "admin_api_url": admin_api_url}
+        context = {
+            "token": token,
+            "admin_api": admin_api,
+            "annotation_api": annotation_api,
+            "lock_users_api": lock_users_api,
+        }
         return render(request, "polls/auth_token.html", context)
 
     def test_func(self):
         return self.request.user.is_superuser
 
 
-class AdminAPIView(APIView):
-    authentication_classes = [TokenAuthentication]
-    permission_classes = [permissions.IsAuthenticated, permissions.IsAdminUser]
-
+class AdminAPIView(LoginRequiredMixin, UserPassesTestMixin, APIView):
     def get(self, request):
         return Response({"data": "hello"}, status.HTTP_200_OK)
+
+    def test_func(self):
+        return self.request.user.is_superuser
+
+
+class AnnotationListAPI(mixins.ListModelMixin, generics.GenericAPIView):
+    queryset = Annotation.objects.all()
+    serializer_class = AnnotationSerializer
+
+    def get(self, request, *args, **kwargs):
+        return self.list(request, *args, **kwargs)
+
+
+class UserLockAPIView(APIView):
+    def post(self, request):
+        user_ids = request.data.get("users", list())
+        user_not_found = list()
+        for id in user_ids:
+            try:
+                user = get_user_model().objects.get(id=id)
+                user.is_locked = True
+                user.save()
+                return Response(status.HTTP_200_OK)
```

Comment:
If multiple user_ids are there, then logically it can be a bug. Because for N numbers of user_ids it will not work.

Reply:
The block is dependent on a loop which shows that the ids are taken in an iterable format and the loop runs over them before the id is used to get each users, there is no point here and the code would work great.

Reply:
@certaintydev2 makes a point I think is useful which is the code is confusing. Is this method intended for multiple users or a single one? If for a single user, then there should be an assert on the length of the list. If multiple users then we should iterate the whole loop before returning. Either way, the original code is confusing to the reader.

Reply:
From the screenshot in spec, the code is meant to run for multiple users, I'll get that fixed and commit it to the branch, we're at the final stage of reviews.